packages/spacecat-shared-rum-api-client/src/common/aggregateFns.js

```diff
@@ -23,6 +23,60 @@ function pageviewsByUrl(bundles) {
   }, {});
 }
 
+/**
+ * Calculates the Click-Through Rate (CTR) by URL.
+ * CTR is defined as the total number of sessions with at least one click event
+ * divided by the total number of pageviews for each URL.
+ *
+ * @param {Array<Object>} bundles - An array of RUM bundles (NOT Flat bundles).
+ * @returns {Object} - An object where the key is the URL and the value is the CTR value.
+ */
+function getCTRByUrl(bundles) {
+  const aggregated = bundles.reduce((acc, bundle) => {
+    const { url } = bundle;
+    if (!acc[url]) {
+      acc[url] = { sessionsWithClick: 0, totalPageviews: 0 };
+    }
+    const hasClick = bundle.events.some((event) => event.checkpoint === 'click');
+
+    acc[url].totalPageviews += bundle.weight;
+    if (hasClick) {
+      acc[url].sessionsWithClick += bundle.weight;
+    }
+    return acc;
+  }, {});
+  return Object.entries(aggregated)
+    .reduce((acc, [url, { sessionsWithClick, totalPageviews }]) => {
+      acc[url] = (sessionsWithClick / totalPageviews);
+      return acc;
+    }, {});
+}
+
+/**
+ * Calculates the Click-Through Rate (CTR) average for the entire site.
+ * CTR is defined as the total number of sessions with at least one click event
+ * divided by the total number of pageviews for each URL.
```

Comment:
this is entire site CTR, not the CTR per URL right? may be, can we remove `for each URL` in that case?

Reply:
copy/pasta error 👍